### PR TITLE
Remove connector factory registrations

### PR DESCRIPTION
**Pull request by majetideepak**
We must register connector factories in the client and not in the Velox library.
The registration of dwrf, parquet, and storage adapters along with HiveConnectorFactory introduces an unnecessary
dependency among these modules.

### CODE REVIEW

**Comment by majetideepak** [comment]
@mbasmanova This work was motivated by our discussion here https://github.com/facebookincubator/velox/pull/8765#discussion_r1504212279 where I was trying to add aliases support for a connector factory.
However, I realized the HiveConnectorFactory has a constructor that accepts a name for the connector factory and therefore, we don't need the aliases support for the HiveConnectorFactory.
But from a Velox library standpoint, we should not be registering the factories in the library. This PR achieves that.

File: velox/connectors/hive/SplitReader.h
Comment (mbasmanova) [inline velox/connectors/hive/SplitReader.h:22]: Why this change?

**Review by mbasmanova — APPROVED** [review]
@majetideepak Looks good to me % one question.

File: velox/connectors/hive/SplitReader.h
Comment (majetideepak) [reply velox/connectors/hive/SplitReader.h]: I reverted this. It was added to fix a build failure from the cleanup. I included the header in the cpp file.

File: velox/dwio/parquet/tests/reader/ParquetTableScanTest.cpp
Comment (czentgr) [inline velox/dwio/parquet/tests/reader/ParquetTableScanTest.cpp:40]: The `registerParquetReaderFactory` also happens in the updated `HiveConnectorTestBase::SetUp` so we don't need it here, right?

File: velox/dwio/parquet/tests/reader/ParquetTableScanTest.cpp
Comment (czentgr) [inline velox/dwio/parquet/tests/reader/ParquetTableScanTest.cpp:44]: The `unregisterParquetReaderFactory` also  occurs in the `HiveConnectorTestBase::TearDown` so we don't need it here?

**Review by czentgr — COMMENTED** [review]
Thank you! Just a few observations.
This would change how externals interact with Velox - as in they need to define the connector factories to be used. So some external users might break with this change.

**Comment by majetideepak** [comment]
@czentgr I addressed your comments. This would be a breaking change. After this lands, we will add the registrations in Prestissimo. I also plan to update the Velox Slack channel.

File: velox/dwio/parquet/tests/reader/ParquetTableScanTest.cpp
Comment (majetideepak) [reply velox/dwio/parquet/tests/reader/ParquetTableScanTest.cpp:39]: fixed.

File: velox/dwio/parquet/tests/reader/ParquetTableScanTest.cpp
Comment (majetideepak) [reply velox/dwio/parquet/tests/reader/ParquetTableScanTest.cpp]: fixed.

**Comment by majetideepak** [comment]
> This would change how externals interact with Velox - as in they need to define the connector factories to be used.

External users have to explicitly register the connector factories. This is similar to any other Velox component. No need to define connector factories.

**Review by czentgr — APPROVED** [review]

**Comment by majetideepak** [comment]
@kgpai Any update on this? Thanks!

**Comment by kgpai** [comment]
Hi Deepak, Please give me a day or so , hoping to land it soon.

**Comment by kevinwilfong** [comment]
Spoke with Krishna and he found this PR breaks Presto Native, he sent details to Deepak offline, I'm going to remove the ready-to-merge tag until that's addressed

**Comment by kgpai** [comment]
@majetideepak Can you fix conflicts , should be good to merge after that..

File: velox/connectors/hive/SplitReader.h
Comment (majetideepak) [reply velox/connectors/hive/SplitReader.h:22]: In my latest rebase, I decided to add this back. SplitReader has the following two fields
```
 std::unique_ptr<dwio::common::Reader> baseReader_;
 std::unique_ptr<dwio::common::RowReader> baseRowReader_;
```
Any class (eg: HiveConnector.cpp) that creates an instance of SplitReader needs to include `#include "velox/dwio/common/Reader.h"` to avoid the error below. It is odd to impose this requirement on the client of 
SplitReader.

```
In file included from /opt/rh/gcc-toolset-12/root/usr/include/c++/12/memory:76,
                 from /usr/local/include/folly/Function.h:205,
                 from /usr/local/include/folly/CancellationToken.h:20,
                 from /__w/velox/velox/./velox/connectors/Connector.h:18,
                 from /__w/velox/velox/./velox/connectors/hive/HiveConnector.h:18,
                 from /__w/velox/velox/velox/connectors/hive/HiveConnector.cpp:17:
/opt/rh/gcc-toolset-12/root/usr/include/c++/12/bits/unique_ptr.h: In instantiation of 'void std::default_delete<_Tp>::operator()(_Tp*) const [with _Tp = facebook::velox::dwio::common::Reader]':
/opt/rh/gcc-toolset-12/root/usr/include/c++/12/bits/unique_ptr.h:396:17:   required from 'std::unique_ptr<_Tp, _Dp>::~unique_ptr() [with _Tp = facebook::velox::dwio::common::Reader; _Dp = std::default_delete<facebook::velox::dwio::common::Reader>]'
/__w/velox/velox/./velox/connectors/hive/SplitReader.h:83:11:   required from 'void std::default_delete<_Tp>::operator()(_Tp*) const [with _Tp = facebook::velox::connector::hive::SplitReader]'
/opt/rh/gcc-toolset-12/root/usr/include/c++/12/bits/unique_ptr.h:396:17:   required from 'std::unique_ptr<_Tp, _Dp>::~unique_ptr() [with _Tp = facebook::velox::connector::hive::SplitReader; _Dp = std::default_delete<facebook::velox::connector::hive::SplitReader>]'
/__w/velox/velox/./velox/connectors/hive/HiveDataSource.h:37:7:   required from 'void std::default_delete<_Tp>::operator()(_Tp*) const [with _Tp = facebook::velox::connector::hive::HiveDataSource]'
/opt/rh/gcc-toolset-12/root/usr/include/c++/12/bits/unique_ptr.h:396:17:   required from 'std::unique_ptr<_Tp, _Dp>::~unique_ptr() [with _Tp = facebook::velox::connector::hive::HiveDataSource; _Dp = std::default_delete<facebook::velox::connector::hive::HiveDataSource>]'
/__w/velox/velox/velox/connectors/hive/HiveConnector.cpp:64:42:   required from here
/opt/rh/gcc-toolset-12/root/usr/include/c++/12/bits/unique_ptr.h:93:23: error: invalid application of 'sizeof' to incomplete type 'facebook::velox::dwio::common::Reader'
   93 |         static_assert(sizeof(_Tp)>0,
      |                       ^~~~~~~~~~~
```

Comment (majetideepak) [reply velox/connectors/hive/SplitReader.h:22]: CI job failed without this change
https://github.com/facebookincubator/velox/actions/runs/10433134539/job/28894546456

**Comment by majetideepak** [comment]
@kgpai this is ready. Can you please import? Thanks.

**Pull request closed without merging**
